### PR TITLE
Remove census algorithm implementation

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes https://codedotorg.atlassian.net/browse/ACQ-410 . Also updates README to reflect the current state of things.

## Testing story

both models touched had extensive unit tests. in each step I removed only the unit tests which depended only on the methods being removed.

## Future work

updating the readme again after ap and ib models + tables are removed is tracked in a separate work item

### CODE REVIEW

File: dashboard/app/models/census/README.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ Each school may have multiple data points for the same school year. All of that
 * **Null** - We did not have any data to process for this school
 
 Historical values are only used when we don't have data for the current year. The summaries are stored in the `census_summaries` table. There should be a row for each school in the `schools` table for each school year that we've computed summaries.
-Summaries are recomputed every night via a [cron job](https://github.com/code-dot-org/code-dot-org/blob/staging/bin/cron/update_census_mapbox) which uploads the current year's summaries to a public MapBox layer and powers the map shown on /yourschool.
+
+As of 2023, the RED team provides the engineering team with a csv file containing the full list of census summaries. The engineering team uses that file to update the public MapBox layer which powers the map shown on /yourschool.
```
Comment (megcrenshaw) [inline dashboard/app/models/census/README.md:54]: 🎉

**Review by megcrenshaw — APPROVED** [review]
Extensive unit tests you weren't kidding! So many 📃 in the 🗑️ 

It looks like we need to keep the `census_submissions` table to determine whether a school teachers CS. Do we need that as a model?

What's the use of keeping the `census_summary` model (and table) in our repo? Is that needed for MapBox?

**Comment by davidsbailey** [comment]
> It looks like we need to keep the `census_submissions` table to determine whether a school teachers CS. Do we need that as a model?
 
Yes, we need this model to get the submission data into the database in response to submissions from the yourschool page:
* https://github.com/code-dot-org/code-dot-org/blob/87d19bd374ad96b875c9295f33735b99a4ecf243/dashboard/app/controllers/api/v1/census/census_controller.rb#L111

CensusYourSchool2017v7 and friends all inherit from CensusSubmission.

> What's the use of keeping the `census_summary` model (and table) in our repo? Is that needed for MapBox?

Great question! We aren't sure yet how we are going to import the census summary info that RED team provides us into the mapbox ([ACQ-385](https://codedotorg.atlassian.net/browse/ACQ-385)). if we can do it directly from csv to mapbox via cronjob, then we shouldn't need the census_summary model or table. this would be the preferred solution. 

I was originally thinking to wait until after we'd implemented ACQ-385 to remove the model and table. I still think we should keep the table, since the data would be hard to recover quickly if all of our data suddenly disappeared from mapbox. however, it does seem like it should be safe to remove the model code now. we can always restore it if we suddenly need to re-upload data to the mapbox to fix a problem. I just created [ACQ-420](https://codedotorg.atlassian.net/browse/ACQ-420) to track removing this model.

**Comment by megcrenshaw** [comment]
All sounds good –– thanks for the context, Dave!

**Pull request merged**